src/commons/utils/include/pdc_malloc.h

```diff
@@ -76,8 +76,8 @@ void *PDC_free(void *mem);
  */
 size_t PDC_get_global_mem_usage();
 
-#define PDC_MALLOC(t)    (t *)PDC_malloc(sizeof(t))
-#define PDC_CALLOC(c, t) (t *)PDC_calloc(c, sizeof(t))
+#define PDC_MALLOC(t)    (t *)PDC_malloc(sizeof(t) + 1)
+#define PDC_CALLOC(c, t) (t *)PDC_calloc(c, sizeof(t) + 1)
```

Comment:
Why +1 here?

Reply:
This was causing the memory segfault in strdup for the tags tests in MacOS. Apparently, there was not enough memory allocated for those which caused the error.

Reply:
- Any clear explanation on why +1 is the fix? 
- Is this a fix only for MacOS? 
- If MacOS needs this +1 fix, what about Linux? 
- Plus, the '+1' fix on calloc call will actually result in 'c' more bytes allocated each time. Any negative impact?

Reply:
We found the origin of the issue. This change has been rolledback.